ui/js/services/tcactions.js

```diff
@@ -1,32 +1,72 @@
 import _ from 'lodash';
 import jsone from 'json-e';
-import { Queue } from 'taskcluster-client-web';
+import { Queue, Auth, Hooks } from 'taskcluster-client-web';
+import { satisfiesExpression } from 'taskcluster-lib-scopes';
 
 import treeherder from '../treeherder';
 import thTaskcluster from './taskcluster';
 
 treeherder.factory('tcactions', [
     '$q', '$http', 'thNotify',
     function ($q, $http, thNotify) {
+        function taskInContext(tagSetList, taskTags) {
+            return tagSetList.some(tagSet =>
+                Object.keys(tagSet).every(
+                    tag => taskTags[tag] && taskTags[tag] === tagSet[tag]
+                )
+            );
+        }
+
         return {
             render: (template, context) => jsone(template, context),
-            submit: ({ action, actionTaskId, decisionTaskId, taskId,
-                      task, input, staticActionVariables }) => {
-
-                const actionTask = jsone(action.task, _.defaults({}, {
+            submit: async ({
+                               action, actionTaskId, decisionTaskId, taskId,
+                               task, input, staticActionVariables
+                           }) => {
+                const context = _.defaults({}, {
                     taskGroupId: decisionTaskId,
                     taskId,
                     task,
```

Comment:
we've also recently removed `task` for kind == hook

Reply:
@djmitche Removing `task` for kind == hook throws an error `unknown context value task`.

Reply:
For [this taskgraph](https://treeherder.mozilla.org/#/jobs?repo=try&revision=d7665b9f894a550d83cfa1bc9ce01fedd11950c6&selectedJob=176466887), [actions.json](https://taskcluster-artifacts.net/ezXvG6gdQIO4CB0lhzLPUg/0/public/actions.json) contains

```
      "hookPayload": {
        "decision": {
          "action": {
            "cb_name": "retrigger_action",
            "description": "Create a clone of the task.",
            "name": "retrigger",
            "symbol": "rt",
            "taskGroupId": "ezXvG6gdQIO4CB0lhzLPUg",
            "title": "Retrigger"
          },
          "parameters": {
            "$eval": "parameters"
          },
          "push": {
            "owner": "mozilla-taskcluster-maintenance@mozilla.com",
            "pushlog_id": "268502",
            "revision": "63d1da26febb2fa51312354a0aa346cbde8839f7"
          },
          "repository": {
            "level": "1",
            "project": "try",
            "url": "https://hg.mozilla.org/try"
          }
        },
        "user": {
          "input": {
            "$eval": "input"
          },
          "taskGroupId": {
            "$eval": "taskGroupId"
          },
          "taskId": {
            "$eval": "taskId"
          }
        }
      },
```

which does not refer to `task`.  Were you using a different (perhaps older) push?